flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBOrderedSetStore.java

```diff
@@ -116,7 +127,14 @@ public RocksDBOrderedSetStore(
 
 	@Override
 	public void add(@Nonnull T element) {
+
 		byte[] elementBytes = serializeElement(element);
+
+		if (LEXICOGRAPIC_BYTE_COMPARATOR.compare(elementBytes, lowerBoundSeekKey) < 0) {
+			// a smaller element means a new lower bound.
+			lowerBoundSeekKey = elementBytes;
```

Comment:
I wonder how could this happen? The `groupPrefixBytes` should be the smallest element since all valid element should start with the `groupPrefixBytes` or did I misunderstand something...